### ISSUE TITLE
Star unpacks as arguments to `cast()` and `assert_type()` are not reported

### ISSUE THREAD

**Issue by InSyncWithFoo** [microsoft/pyright#9674]
Minimal reproducible example ([playground](https://pyright-play.net/?pyrightVersion=1.1.391&pythonVersion=3.14&strict=true&enableExperimentalFeatures=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoCqKCAhgMYDWANFCQM50CmeA%2BvAozWfTALABQAgdzowAFACoYAVwQAbRgG1RIGqhgBdGgDpdASiiGjxk1ADEUFGCjNwIYTzFFSlRTPlKValJq1Rd2gamhha2uABchMTkFJh0lmD4JHJyYADujAAmmCiwABZI8WRgPowAHnz8IuISEplIZDDKMKo5mjr6wV3mNiB2kQBEAQNxCUkp6Vk5%2BYVQxaUVDqJO0a71jc2t6hp%2BAUFdoX0RUS6xs1bjqRnZqDNFJTDllQL0TKzsjGIBNFKyCpveTT7UyHfo2MocRpTAbuf5eNoaEYAI2k%2BBAjDIjCQADdoQBRFJIBB0QoAFUQjAGLwYzBgbApX10NGcMTcf08LUBOyCoOOLMoowutAm12mMAK9wWlVetPpHEZ2h%2BdQaTXh22BJl5IEi5Uhj2yA3Wqs5CORqKg6MxOPxhOJZIpVP4MveDO%2BJ1ZRoBCI0%2By1kX5Z3iQuSVymt3Fs3mj0W-CAA)):

```python
cast(*tuple[str, int], ...)                # no error
cast(Unpack[tuple[str, int]], ...)         # error: Unpack is not allowed in this context
cast(**dict[str, int], ...)                # error: "..." is not allowed in this context
cast(Unpack[dict[str, int]], ...)          # error: Unpack is not allowed in this context

assert_type(..., *tuple[str, int])         # error: expected "tuple[str, int]" but received "EllipsisType"
assert_type(..., Unpack[tuple[str, int]])  # error: Unpack is not allowed in this context
assert_type(..., **dict[str, int])         # error: expected "dict[str, int]" but received "EllipsisType"
assert_type(..., Unpack[dict[str, int]])   # error: Unpack is not allowed in this context
```

`Unpack`-ed types are not valid in these contexts and should be reported whatever forms they take.

**Comment by erictraut** [comment]
There are two separate bugs here. One is a missing syntax error check for the case where a positional argument follows an unpacked keyword argument in a call expression (as in `cast(**dict[str, int], ...)`), and the other missing check is specific to the first positional argument of a `cast` call. Both of these will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.392.
